src/storage_common.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) The ThingSet Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <thingset.h>
+#include <thingset/sdk.h>
+#include <thingset/storage.h>
+
+LOG_MODULE_REGISTER(thingset_storage_common, CONFIG_THINGSET_SDK_LOG_LEVEL);
+
+static struct k_work_delayable storage_work;
+
+static bool storage_save_inhibit = IS_ENABLED(CONFIG_THINGSET_STORAGE_INHIBIT_OVERWRITE);
```

Comment:
The negative name (`inhibit`) means that the ideal case is `inhibit == false`, a double negative, which is not easy to read. `storage_save_enabled` would be clearer, even if the semantics are slightly different.